### PR TITLE
User auth

### PR DESCRIPTION
**Pull request by hoalex21**

### CODE REVIEW

File: backend/src/server.ts [hunk 1]
```diff
@@ -1,17 +1,140 @@
...
+  }
+
+  if (errors.length > 0) {
+    res.status(400).send({
+      errors: errors
+    })
+  } else {
+    const loginToken = makeToken()
+
+    tokenStorage["login"] = loginToken
```
Comment (devappslabs) [inline backend/src/server.ts]: does this work for multiple users logging in? don't multiple users update the same "login" key of the global tokenStorage here? I would think the key we update is the loginToken string

Comment (devappslabs) [reply backend/src/server.ts]: may need to update other related code

**Review by devappslabs — CHANGES_REQUESTED** [review]

**Review by devappslabs — CHANGES_REQUESTED** [review]
use `serverUrl` constant I created in frontend code so it's easy to change later if needed (ex. during deployment)

File: backend/src/server.ts (hunk 1 above)
Comment (hoalex21) [reply backend/src/server.ts]: done

**Review by devappslabs — APPROVED** [review]

**Pull request merged**